packages/common/test/util.test.ts

```diff
@@ -675,4 +678,40 @@ describe("util functions", () => {
       expect(chunkArray([], 3)).toEqual([]);
     });
   });
+  describe("last: ", () => {
+    it("returns last element of an array", () => {
+      expect(last([1, 2, 3])).toBe(3, "should return last entry");
+      const objArr = [{ id: 1 }, { id: 2 }, { id: 3 }];
+      expect(last(objArr)).toBe(objArr[2], "returns last object from an array");
+      expect(last([])).toBeUndefined();
+    });
+  });
+  describe("filterBy:", () => {
+    it("filters array by deep prop value", () => {
+      const arr = [
+        { item: { type: "Map", id: "map1" } },
+        { item: { type: "App", id: "app1" } },
+        { item: { type: "Map", id: "map2" } },
+        { item: { type: "App", id: "app2" } }
+      ];
+      const apps = filterBy(arr, "item.type", "App");
+      expect(apps.length).toBe(2, "should have two apps");
+      apps.forEach(app => {
+        expect(app.item.type).toBe("App");
+      });
+    });
+  });
+
+  describe("uniqueBy: ", () => {
+    it("returns uniqu entries by deep prop", () => {
+      const arr = [
+        { item: { type: "Map", id: "map1" } },
+        { item: { type: "App", id: "app1" } },
+        { item: { type: "Map", id: "map1" } },
+        { item: { type: "App", id: "app2" } }
+      ];
+      const chk = uniqueBy(arr, "item.id");
```

Comment:
seems like `uniqueBy(arr, "item.type")` would be a better example just b/c we expect id to be unique across all items.

Reply:
I'll mull this - how I use it, and why I wrote it, was actually to unique a set of `IDependency` objects, by Item id, so we don't try to convert the same item multiple times... but yeah - in general we'd not expect to have dupes of item-ish things